### PR TITLE
Add fixMath lib

### PR DESCRIPTION
**Pull request by yusufidimaina9989**

### CODE REVIEW

File: src/fixMath.ts [hunk 1]
```diff
@@ -0,0 +1,386 @@
+import { SmartContractLib, and, assert, lshift, method, prop, rshift } from "scrypt-ts";
+
+export class FixMath extends SmartContractLib{
```
Comment (msinkec) [inline src/fixMath.ts:3]: Can you fix the strange indents? (CTRL + Shift + I in VScode should do it)

**Comment by yusufidimaina9989** [comment]
https://github.com/sCrypt-Inc/scrypt-ts-lib/issues/18

File: tests/contracts/fixMathTest.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    let res: bigint = FixMath.log(402139020806868238336n);
+    assert(res == 56851204471023106498n);
+  }
+
+  @method()
+  public testExp() {
+    let exptest1 = FixMath.exp(10n * TestFixMath.scale);
+    assert(exptest1 / TestFixMath.scale == 22026n);
+  }
+}
```
Comment (msinkec) [inline tests/contracts/fixMathTest.ts:21]: Run these in a test file to make sure they actually pass the assertions.

File: tests/contracts/txParserBTCTest.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ import {
     FixedArray,
 } from 'scrypt-ts'
 
-export class TransactionParserTest extends SmartContract {
+export class TransactionParserTestBTC extends SmartContract {
```
Comment (msinkec) [inline tests/contracts/txParserBTCTest.ts:17]: I don't think this should be part of this PR. Revert please.

File: tests/specs/txParserBTC.test.ts [hunk 4]
```diff
@@ -1,15 +1,15 @@
 import { expect } from 'chai'
```
Comment (msinkec) [inline tests/specs/txParserBTC.test.ts:1]: Same for this file.

**Review by msinkec — CHANGES_REQUESTED** [review]

File: tests/contracts/txParserBTCTest.ts (hunk 3 above)
Comment (yusufidimaina9989) [reply tests/contracts/txParserBTCTest.ts:17]: There is another class with same name that is why the CI is failling, that is the reason i rename it

**Pull request merged**